diff --git a/src/Data/Strict/Map/Autogen/Internal.hs b/src/Data/Strict/Map/Autogen/Internal.hs
index e5303ca..949b4f6 100644
--- a/src/Data/Strict/Map/Autogen/Internal.hs
+++ b/src/Data/Strict/Map/Autogen/Internal.hs
@@ -458,7 +458,7 @@ m1 \\ m2 = difference m1 m2
 -- their union @m1 <> m2@ maps @k@ to @a1@.
 
 -- See Note: Order of constructors
-data Map k a  = Bin {-# UNPACK #-} !Size !k a !(Map k a) !(Map k a)
+data Map k a  = Bin {-# UNPACK #-} !Size !k !a !(Map k a) !(Map k a)
               | Tip
 
 type Size     = Int
